catalyst-gateway/bin/src/metrics/chain_follower/mod.rs

```diff
@@ -1,65 +1,35 @@
 //! Metrics related to Chain Follower analytics.
 
-use std::{
-    sync::atomic::{AtomicBool, Ordering},
-    thread,
-};
-
 use cardano_blockchain_types::Network;
 use cardano_chain_follower::Statistics;
 
 use crate::settings::Settings;
 
 mod reporter;
 
-/// This is to prevent the init function from accidentally being called multiple times.
-static IS_INITIALIZED: AtomicBool = AtomicBool::new(false);
-
-/// Starts a background thread to periodically update Chain Follower metrics.
-///
-/// This function spawns a thread that updates the Chain Follower metrics
-/// at regular intervals defined by `METRICS_FOLLOWER_INTERVAL`.
-pub(crate) fn init_metrics_reporter() {
-    if IS_INITIALIZED.swap(true, Ordering::SeqCst) {
-        return;
-    }
-
+/// Updates Chain Follower metrics to current values.
+pub fn update() {
```

Comment:
```suggestion
pub(crate) fn update_metrics() {
```

Reply:
It a more like a convention (not a strict one) which comes more than a year ago, to be precise with the scope.
Yea I agree that in this case it does not make to much sense, but in the scenario if such code could be also shared as a library one, better at the beginning defining what we really want to make public for everyone.